.azure-pipelines/buildAndPackage.yml

```diff
@@ -30,10 +30,62 @@ pr: none
 pool:
   vmImage: windows-latest
 
-steps:
-- template: templates/checkout-and-credscan.yml
-- template: templates/install-java.yml
-- template: templates/secure-files.yml
-- template: templates/build-and-coverage.yml
-- template: templates/publish-artefacts.yml
-- template: templates/alert-failure.yml
+stages:
+- stage: Build
+  jobs:
+  - job: Build
+    steps:
+    - template: templates/build-stage/checkout-and-credscan.yml
+    - template: templates/build-stage/install-java.yml
+    - template: templates/build-stage/secure-files.yml
+    - template: templates/build-stage/build-and-coverage.yml
+    - template: templates/build-stage/publish-artefacts.yml
+    - template: templates/alert-failure.yml
+
+- stage: Maven-Preview
+  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/dev'))
+  jobs: 
+  - job: Maven-Preview
+    steps: 
+    - template: templates/download-artifacts.yml
+    - template: templates/maven-preview-stage/secure-files.yml
+    - template: templates/maven-preview-stage/copy-and-build.yml
+    - template: templates/alert-failure.yml
+
+- stage: Maven-Release-And-Github-Release
+  dependsOn: []
+  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+  jobs:
+  - deployment: Maven-Release
+    environment: 'MSGraph SDK Development - Java'
+    pool:
+      vmImage: windows-latest
+    strategy:
+     runOnce:
+      deploy:
+        steps:
+        - template: templates/maven-release-stage/secure-files.yml
+        - template: templates/maven-release-stage/copy-and-build.yml
+        - template: templates/alert-failure.yml
+
+  - deployment: Github-Release
+    dependsOn: Maven-Release
+    environment: 'MSGraph SDK Development - Java'
+    pool:
+      vmImage: windows-latest
+    strategy: 
+     runOnce:
+      deploy:
+        steps: 
+        - template: templates/github-release-stage/version-and-release.yml
+        - template: templates/alert-failure.yml
+
+#- stage: Github-Release
```

Comment:
cleanup

Reply:
Kept the commented out code in case we wanted to revert, however if having the Github release as a deployment job is optimal then I'll go ahead and clear the commented out code.